signbank/settings/base.py

```diff
@@ -8,8 +8,12 @@
 import sys
 
 import dj_database_url
-from django.utils.translation import ugettext_lazy as _
+from django.utils.translation import gettext_lazy as _
 
+# Workaround for a Django breaking change in 4+
+import django
+from django.utils.encoding import smart_str
+django.utils.encoding.smart_text = smart_str
```

Comment:
question: can you explain this more?

Why do we need to do this? what do we need to do to not do it? etc

Reply:
An internal Django library we have no control over still uses `smart_text`, or did during the upgrade.
I'll check if this is still the case.

Reply:
Still the case. It's a python 3.9 issue:

```
#0 0.414   File "/app/signbank/dictionary/models.py", line 17, in <module>
#0 0.414     from tagging.models import Tag
#0 0.414   File "/usr/local/lib/python3.9/site-packages/tagging/models.py", line 8, in <module>
#0 0.414     from django.utils.encoding import smart_text
#0 0.414 ImportError: cannot import name 'smart_text' from 'django.utils.encoding' (/usr/local/lib/python3.9/site-packages/django/utils/encoding.py)


```

Reply:
> It's a python 3.9 issue

No it's not 🙂 

> An internal Django library we have no control over still uses smart_text

I will always bet good money that it'll never be an issue in an internal django library, given you're upgrading django 🙂 

---

The error tells you the library at fault: 

```
#0 0.414   File "/app/signbank/dictionary/models.py", line 17, in <module>
#0 0.414     from tagging.models import Tag
#0 0.414   File "/usr/local/lib/python3.9/site-packages/tagging/models.py", line 8, in <module>
#0 0.414     from django.utils.encoding import smart_text
```

The first and second line show you our own code which already points at the smoking gun: `from tagging.models import Tag`. The third line then causes the gun to smoke even more: `File "/usr/local/lib/python3.9/site-packages/tagging/models.py"`.

So that leads us to understand [`django-tagging`](https://pypi.org/project/django-tagging/) is the troublesome library and sure enough https://github.com/Fantomas42/django-tagging/pull/23

As a point for the future, this is the kind of thing we should be tracking down as part of these upgrades - I'm ok with your solution here for the short-term as a way of unblocking this upgrade, but we should have a comment that clearly explains what the issue is and what we might want to do to address it because these are one of the most literal examples of "technical debt" that you'll find 🙂 

i.e. in this case, we've found that we're using a library which is no longer being maintained so we might want to consider replacing it with something else or seeing if we can remove it all together.

---

The actual action here is for you to update the comment explaining this is a workaround for an issue in `django-tagging` which seems to no longer be maintained, etc etc...

Reply:
Right, took me a while to figure out what you were actually saying! 😆 
I'll duly note.

Reply:
Done.